common/playlist.c

```diff
@@ -178,6 +179,27 @@ void playlist_shuffle(struct playlist *pl)
     talloc_free(arr);
     pl->current = save_current;
     pl->current_was_replaced = save_replaced;
+    pl->shuffled = true;
+}
+
+void playlist_unshuffle(struct playlist *pl) {
+    struct playlist_entry *save_current = pl->current;
+    bool save_replaced = pl->current_was_replaced;
+    int count = playlist_count(pl);
+    struct playlist_entry **arr = talloc_array(NULL, struct playlist_entry *,
+                                               count);
+    for (int n = 0; n < count; n++) {
+        arr[pl->first->pos] = pl->first;
```

Comment:
That could just do invalid memory accesses and crash under certain circumstances.